### ISSUE TITLE
Wrong indentation of conditional compilation blocks around switch cases

### ISSUE THREAD

**Issue by fphilipe** [nicklockwood/SwiftFormat#234]
Swift allows to wrap a switch case in a conditional compilation block. SwiftFormat indents those blocks one level too deep.

This is what I get for each config of `--ifdef`:

```swift
// indent:
func abc() {
    switch foo {
    case .bar:
        return 123
        #if FLAG
            case .baz:
                return 456
        #endif
    }
}

// noindent:
func abc() {
    switch foo {
    case .bar:
        return 123
        #if FLAG
        case .baz:
            return 456
        #endif
    }
}

// outdent:
func abc() {
    switch foo {
    case .bar:
        return 123
#if FLAG
        case .baz:
            return 456
#endif
    }
}
```

What I expect is the following:

```swift
// indent:
func abc() {
    switch foo {
    case .bar:
        return 123
    #if FLAG
        case .baz:
            return 456
    #endif
    }
}

// noindent:
func abc() {
    switch foo {
    case .bar:
        return 123
    #if FLAG
    case .baz:
        return 456
    #endif
    }
}

// outdent:
func abc() {
    switch foo {
    case .bar:
        return 123
#if FLAG
    case .baz:
        return 456
#endif
    }
}
```

**Comment by nicklockwood** [comment]
That makes sense. Thanks for reporting!

**Comment by nicklockwood** [comment]
@fphilipe this turns out to be harder to fix than I expected. Didn't make it into this release, but hopefully will be solved for the next one.

**Comment by nicklockwood** [comment]
@fphilipe fixed in version 0.33.8
